rust/core-lib/src/file.rs

```diff
@@ -266,6 +275,17 @@ fn get_mod_time<P: AsRef<Path>>(path: P) -> Option<SystemTime> {
     File::open(path).and_then(|f| f.metadata()).and_then(|meta| meta.modified()).ok()
 }
 
+/// Returns the SHA256 checksum of a file at a given path, if present.
+fn get_sha256<P: AsRef<Path>>(path: P) -> Option<Vec<u8>> {
+    let mut hasher = Sha256::new();
+    let mut output = Vec::new();
+    File::open(path)
+        .and_then(|mut f| std::io::copy(&mut f, &mut hasher))
+        .map(|_| output.copy_from_slice(hasher.result().as_slice()))
+        .map(|_| output)
```

Comment:
this is weird... we're sort of hijacking the map to return something external. I would prefer,

```rust
File::open(path)
         .and_then(|mut f| std::io::copy(&mut f, &mut hasher))
         .map(|_| output.copy_from_slice(hasher.result().as_slice()))?;
Some(output)
```

but I'm not totally sure what the implication of failure is, here?

Reply:
I don't really want to return a `Result<Option<_>, Error>` here :/

Reply:
well okay, but what if the file exists and we can't open it? In general I'm suspicious of discarding errors. If this file doesn't exist, can we handle that earlier, and only call this when we know we have a file? That would let us return a `Result` instead of an option.

Reply:
Sure, I just tried to mimic get_mod_time here, I guess we should change it too? Anyway, I'm currently on holidays, so that'll take me bit :)

Reply:
Cool, enjoy!

Reply:
Is this the only blocker to this?

Reply:
Kind of. Will work on this later today. If you feel like helping: Could you test if Xi sends you an alarm WITHOUT this PR when you open a file, then change it and then try to save to it? It doesn't for me, which suggests that modtime doesn't work for me for some reason :c

Reply:
@Cogitri, I don't know exactly what you mean by alarm, but xi seems to be working fine without this PR.

Reply:
alert is a message from xi which usually opens an error dialogue in your frontend.